src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPatchHandler.cs

```diff
@@ -177,12 +179,17 @@ public class VmrPatchHandler : IVmrPatchHandler
 
         // If current mapping hosts VMR's non-src/ content, synchronize it too
         // We only do it when processing the root mapping, not its submodules
-        var sourcesPath = _vmrInfo.GetRepoSourcesPath(mapping);
-        if (relativePath == mapping.Name && (_vmrInfo.ContentPath?.StartsWith(sourcesPath) ?? false))
+        var relativeRepoPath = _vmrInfo.GetRelativeRepoSourcesPath(mapping);
+        int i = 1;
+        foreach (var (source, destination) in _vmrInfo.AdditionalMappings.Where(m => m.Source.StartsWith(relativeRepoPath)))
         {
-            _logger.LogInformation("Mapping {mapping} contains VMR's non-src/ content. Creating patch for it too..", mapping.Name);
+            var relativeClonePath = source.Substring(relativeRepoPath.Length + 1);
+
+            _logger.LogInformation("Detected 'non-src/' mapped content in {source}. Creating patch..", source);
+
+            patchName = $"{(destination != null ? destination.Replace('/', '_') : "root")}-{Commit.GetShortSha(sha1)}-{Commit.GetShortSha(sha2)}-{i++}.patch";
```

Comment:
Wouldn't including `source` in the patch name produce a better unique patch name then a counter? Or would it be too messy (even if we were to include just the end folder of the `source` path)

Reply:
Yes, you're right and I had that originally, but it's getting very long already (source is this at the moment: `src/installer/SourceBuild/tarball/content`)

Reply:
I guess as long as one destination does not have a lot of patches with the same commit sha's we should be A-OK